Update dependencies

jurgenwerk opened the pull request:
Updated a couple of dependencies where `yarn audit` reported vulnerabilities.

jurgenwerk commented:
Looking into the lint issue...

jurgenwerk commented on .solhint.json:
```diff
@@ -8,6 +8,7 @@
     "no-complex-fallback": "off",
-    "compiler-version": ["error", "0.4.24"]
+    "compiler-version": ["error", "0.4.24"],
+    "reason-string": "off"
```
https://protofire.github.io/solhint/docs/rules/best-practises/reason-string.html

There's quite a lot of contracts where the reason is not provided I opted into turning this warning off

alex-cardstack reviewed (approved):
Looks good, thanks!

alex-cardstack commented on contracts/upgradeable_contracts/BasicHomeBridge.sol:
```diff
@@ -22,7 +22,7 @@ contract BasicHomeBridge is EternalStorage, Validatable, BasicBridge, BasicToken
         address authorityResponsibleForRelay,
         bytes32 messageHash,
-        uint256 NumberOfCollectedSignatures
+        uint256 numberOfCollectedSignatures
```
Events are processed externally, we should not change the name here as it will result in inconsistent history on-chain

alex-cardstack commented on contracts/upgradeable_contracts/arbitrary_message/BasicHomeAMB.sol:
```diff
@@ -12,7 +12,7 @@ contract BasicHomeAMB is BasicAMB, MessageDelivery {
         address authorityResponsibleForRelay,
         bytes32 messageHash,
-        uint256 NumberOfCollectedSignatures
+        uint256 numberOfCollectedSignatures
```
Events are processed externally, we should not change the name here as it will result in inconsistent history on-chain

jurgenwerk replied:
I see! Fixed in https://github.com/cardstack/tokenbridge-contracts/pull/12/commits/f739bb854afc2852a9f949d05b8f19bfb3e77f4c

alex-cardstack commented on contracts/upgradeable_contracts/erc721_to_erc721/ERC721Proxy.sol:
```diff
@@ -38,9 +38,8 @@ contract ERC721Proxy is Proxy {
-    * @param _chainId chain id for current network.
     */
-    constructor(string memory _name, string memory _symbol, uint256 _chainId) public {
+    constructor(string memory _name, string memory _symbol) public {
```
Hmm, I think we should be storing the chainId instead of removing the param but I might have to review a bit more why it's not used right now

jurgenwerk replied:
Ok, so `chainId = _chainId;`? Why do you think we need it - what would it be used for?

alex-cardstack replied:
@jurgenwerk this is a real bug and it's great that we've caught it before we deployed any bridged NFTs!

I actually only realised this when looking at the chainId stuff as part of CS-2862 - I will address as part of that issue

alex-cardstack replied:
@jurgenwerk this is part of the code we're going to delete - so fine to roll it back for now and I'll address it as part of chainId work

alex-cardstack commented on contracts/upgradeable_contracts/erc721_to_erc721/mediator/HomeNftMediator.sol:
```diff
@@ -76,7 +76,7 @@ contract HomeNftMediator is BasicNftMediator, IHomeNftMediator {
         symbol = string(abi.encodePacked(symbol, ".CPXD"));
 
-        homeTokenAddress = new ERC721Proxy(name, symbol, bridgeContract().sourceChainId());
+        homeTokenAddress = new ERC721Proxy(name, symbol);
```
I think we should be passing this in and using it in the contract but need to confirm for sure

jurgenwerk commented:
@alex-cardstack thanks for clarification, I removed the commit where `_chainId` was removed, and added a temporary linter ignore line until the chain id validation work is done.
